### ISSUE TITLE
Extension development host crash when using child_process.exec(Sync)

### ISSUE THREAD

**Issue by MrAndersen1** [microsoft/vscode#138036]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.62.3
- OS Version: macOS Big Sur 11.6

Steps to Reproduce:
When developing an extension I ran into a hard crash of the extension development host when running a command with ```execSync``` with an invalid directory in ```cwd```, even with a try-catch block around it.
When running with ```exec``` instead, or without ```stdio: inherit```, I can see an error in the developer tools at least.
While the fault is mine, with the invalid directory, it should probably not hard crash.

```typescript
vscode.commands.registerCommand("myCommand", async () => {
	try{
		// Using any one of these three lines gives different errors
		// this hard-crashes the extension development host
		execSync(`pwd`, {"cwd": "/Users/me/nonExistingDir", "stdio": "inherit"}); 

		// without "inherit" we get error nr 1
		execSync(`pwd`, {"cwd": "/Users/me/nonExistingDir"});

		// exec instead of execSync gives error nr 2
		exec(`pwd`, {"cwd": "/Users/me/nonExistingDir"});
	} catch(e) {
		console.error(e);
	}
})
```

Hopefully this is enough for you to be able reproduce it with. If not then I'll gladly help with whatever more is required.

In addition to this, when trying this out the window I develop the extension in starts showing
```
ERR [Extension Host] (node:2838) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 uncaughtException listeners added to [process]. Use emitter.setMaxListeners() to increase limit
```
but this is just a consequence I'm guessing.

error 1:
```
ERR [Extension Host] Error: spawnSync /bin/sh ENOTCONN
	at Object.spawnSync (internal/child_process.js:1067:20)
	at spawnSync (child_process.js:619:24)
	at Object.execSync (child_process.js:664:15)
	at Object.execSync (electron/js2c/asar_bundle.js:5:12288)
	at /Users/me/myExtension/out/extension.js:247:29
	at Generator.next (<anonymous>)
	at /Users/me/myExtension/out/extension.js:8:71
	at new Promise (<anonymous>)
	at /Users/me/myExtension/out/extension.js:4:12
	at /Users/me/myExtension/out/extension.js:236:75
	at o._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:94:111751)
	at o.$executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:94:112293)
	at n._doInvokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:13802)
	at n._invokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:13486)
	at n._receiveRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:12148)
	at n._receiveOneMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:10826)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:8922
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)
	at s.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:15990)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:113:30625
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)
	at s.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:15990)
	at f._receiveMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:21260)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:18138
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)
	at w.acceptChunk (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:13351)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:12699
	at Socket.m (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:113:13575)
	at Socket.emit (events.js:315:20)
	at addChunk (internal/streams/readable.js:309:12)
	at readableAddChunk (internal/streams/readable.js:284:9)
	at Socket.push (internal/streams/readable.js:223:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:188:23)
	at Pipe.callbackTrampoline (internal/async_hooks.js:131:14)
```

error 2
```
[Extension Host] RangeError [ERR_OUT_OF_RANGE]: The value of "err" is out of range. It must be a negative integer. Received 1
	at Object.getSystemErrorName (util.js:231:11)
	at errnoException (internal/errors.js:501:21)
	at ChildProcess.spawn (internal/child_process.js:403:11)
	at spawn (child_process.js:570:9)
	at Object.execFile (child_process.js:246:17)
	at Object.execFile (electron/js2c/asar_bundle.js:5:2242)
	at Object.exec (child_process.js:167:25)
	at Object.exec (electron/js2c/asar_bundle.js:5:12288)
	at /Users/me/myExtension/out/extension.js:246:29
	at Generator.next (<anonymous>)
	at /Users/me/myExtension/out/out/extension.js:8:71
	at new Promise (<anonymous>)
	at /Users/me/myExtension/out/extension.js:4:12
	at /Users/me/myExtension/out/extension.js:236:75
	at o._executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:94:111751)
	at o.$executeContributedCommand (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:94:112293)
	at n._doInvokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:13802)
	at n._invokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:13486)
	at n._receiveRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:12148)
	at n._receiveOneMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:10826)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:98:8922
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)
	at s.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:15990)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:113:30625
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)
	at s.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:15990)
	at f._receiveMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:21260)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:18138
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:66:1712)
	at w.acceptChunk (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:13351)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:74:12699
	at Socket.m (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:113:13575)
	at Socket.emit (events.js:315:20)
	at addChunk (internal/streams/readable.js:309:12)
	at readableAddChunk (internal/streams/readable.js:284:9)
	at Socket.push (internal/streams/readable.js:223:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:188:23)
	at Pipe.callbackTrampoline (internal/async_hooks.js:131:14)
```

**Comment by weinand** [comment]
Yes, I can reproduce this.

**Comment by deepak1556** [comment]
This is fixed in latest stable `1.73`
